tt_metal/hw/inc/noc_common.h

```diff
@@ -0,0 +1,14 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#ifndef _NOC_COMMON_H_
+#define _NOC_COMMON_H_
+
+// Helper functions to convert NoC coordinates to NoC-0 coordinates, used in metal as "physical" coordinates.
+#define NOC_0_X(noc_index, noc_size_x, x) (noc_index == 0 ? (x) : (noc_size_x-1-(x)))
```

Comment:
let's put this (duplicated) in noc_nonblocking_api.h, is really device specific (even though it is the same across current arches)